ffig/generators/go.py

```diff
@@ -4,10 +4,39 @@
 # for generator plugins. Ultimately, it will generate Go packages in the
 # correct structure.
 
+import os
+import os.path
+
 import generators
+import logging
+
+log = logging.getLogger(__name__)
 
 def go_generator(binding, api_classes, env, args, output_dir):
-    return generators.default_generator(binding, api_classes, env, args, output_dir)
+    ''' Create Go bindings in an appropriate Go package.
+
+        This generator produces a directory src/modulename inside the output
+        directory, into which it writes a single Go file containing the
+        bindings for this module.
+    '''
+    module_directory = os.path.realpath(os.path.join(output_dir, 'src', args.module_name))
+    if not os.path.isdir(module_directory):
+        log.info('Creating Go package directory {}'.format(module_directory))
+        os.makedirs(module_directory)
+
+    log.info('Generating Go bindings for module {0}'.format(args.module_name))
+    template = env.get_template(binding)
+    generated_code = generators.render_api_and_obj_classes(api_classes, template)
+
+    output_file_name = os.path.join(module_directory,
+            generators.get_template_output(args.module_name,
+                generators.get_template_name(binding)))
+    with open(output_file_name, 'w') as f:
+        f.write(generated_code)
+        log.info('Wrote Go bindings for module {0} to {1}'.format(
+            args.module_name, output_file_name))
+
+    return output_file_name
```

Comment:
Perhaps they should return a list of filenames. I'm thinking that (part of) the REST API might work by creating temporary files in the right directory structure and archiving them. For that, a list of generated files could be useful.

But nothing uses it yet...